### PR TITLE
File attachments SDK improvements

### PR DESCRIPTION
**Pull request by richardgill**
Had a go at rewriting this docs page.

### CODE REVIEW

File: 040-SDK/150-file-attachments.mdx [hunk 1]
```diff
@@ -9,23 +9,37 @@ published: true
...
 
-- [Upload URLs](#upload-urls), when requested from an existing file object, provide a temporary, secure URL to upload files to. Because they don't require an API key they can be used client-side and are good when dealing with large files through a client interface like a browser. This method comes at the downside of requiring extra steps across the client and server side.
-- [Record APIs](#record-apis) provide the easiest way to upload files, and are the only methods that allow uploading files during the same step as creating a record. The downside is that they only support file payloads under 20MB and will likely run over the limits of serverless functions from providers like Vercel which can be much lower.
-- [Binary APIs](#file-binary-apis) allow for uploading large files on an existing record. Unfortunately, unlike Upload URLs, they must be called server side, which will likely run over the limits of serverless functions from providers like Vercel. This makes their usefulness limited to scenarios where you control the environment completely, like uploading large files from a local machine.
+[Upload URLs](#upload-urls) (recommended)
```
Comment (richardgill) [inline 040-SDK/150-file-attachments.mdx:12]: @snide I'd like these to be a bit more headery

File: 040-SDK/150-file-attachments.mdx [hunk 2]
```diff
@@ -9,23 +9,37 @@ published: true
...
 
-Xata recommends using Upload URLs for most scenarios. Record and Binary APIs exist for more specific situations where you know the size of your data or have complete control over the hosting infrastructure.
+- Client side (don't require using an API key)
```
Comment (richardgill) [inline 040-SDK/150-file-attachments.mdx:14]: I'd like ✅  and ⚠️  here or something

File: 040-SDK/150-file-attachments.mdx [hunk 3]
```diff
@@ -39,55 +55,40 @@ const record = await xata.db.myTableName.create(
...
+  });
+  if (response.status !== 200) {
+    throw new Error("Couldn't create record");
+  }
 
-    // The response include a pre-signed uploadUrl on the record. Below, we then send a file
-    // directly to Xata on the client side using the PUT request. This lets us upload
-    // large files that would otherwise exceed the limit for serverless functions on
-    // services like Vercel.
+  const record = await response.json();
```
Comment (richardgill) [inline 040-SDK/150-file-attachments.mdx:77]:
```suggestion
  const responseJson = await response.json();
```

File: 040-SDK/150-file-attachments.mdx [hunk 4]
```diff
@@ -39,55 +55,40 @@ const record = await xata.db.myTableName.create(
...
-    } else {
-      throw new Error("Couldn't upload image because the record wasn't created");
-    }
+  try {
+    // Put the file inside a FormData object
+    const formData = new FormData();
+    const fileObj = file;
+    formData.append('fileType', fileObj.type);
+    // Use `myUploadUrl` from the server response to upload the file on the client
+    await fetch(record.myUploadUrl, { method: 'PUT', body: file });
```
Comment (richardgill) [inline 040-SDK/150-file-attachments.mdx:85]:
```suggestion
    await fetch(responseJson.myUploadUrl, { method: 'PUT', body: file });
```

File: 040-SDK/150-file-attachments.mdx [hunk 5]
```diff
@@ -2,30 +2,46 @@
 title: File attachments
 navTitle: File attachments
 keywords: ['files', 'file storage', 'file attachments']
-description: Xata effectively keeps track of files within a database
+description: Store files within your database with Xata
```
Comment (richardgill) [inline 040-SDK/150-file-attachments.mdx:5]: @snide I changed this so it shows up in the diff. Any better ideas?

Comment (snide) [reply 040-SDK/150-file-attachments.mdx:5]: That's fine.

**Review by snide — APPROVED** [review]
LGTM, let's merge it.

**Pull request merged**